semantics/src/main/scala/aqua/semantics/expr/ServiceSem.scala

```diff
@@ -27,44 +27,45 @@ class ServiceSem[S[_]](val expr: ServiceExpr[S]) extends AnyVal {
     T: TypesAlgebra[S, Alg],
     V: ValuesAlgebra[S, Alg],
     D: DefinitionsAlgebra[S, Alg]
-  ): EitherT[Alg, Raw, ServiceRaw] = for {
-    arrows <- EitherT.fromOptionF(
-      // TODO:  Move to purgeDefs here, allow not only arrows
-      //        from parsing, throw errors here
-      D.purgeArrows(expr.name),
-      Raw.error("Service has no arrows")
-    )
-    arrowsByName = arrows.map { case (name, arrow) =>
-      name.value -> (name, arrow)
-    }.toNem
-    defaultId <- expr.id.traverse(id =>
-      EitherT.fromOptionF(
-        V.valueToStringRaw(id),
-        Raw.error("Failed to resolve default service id")
+  ): EitherT[Alg, Raw, ServiceRaw] = {
+    def errored = ErroredPart(expr.name.value)
+    for {
```

Comment:
If error is always the same, this whole `for` could be rewritten to `for` on `OptionT` and converted to `EitherT` at the very end.